src/ImageSharp/PixelFormats/PixelOperations{TPixel}.cs

```diff
@@ -61,33 +60,169 @@ public partial class PixelOperations<TPixel>
+            Utils.Vector4Converters.Default.DangerousFromScaledVector4(sourceVectors, destPixels);
         }
 
         /// <summary>
         /// Bulk version of <see cref="IPixel.ToScaledVector4()"/> converting 'sourceColors.Length' pixels into 'destinationVectors'.
         /// </summary>
         /// <param name="configuration">A <see cref="Configuration"/> to configure internal operations</param>
-        /// <param name="sourceColors">The <see cref="Span{T}"/> to the source colors.</param>
-        /// <param name="destinationVectors">The <see cref="Span{T}"/> to the destination vectors.</param>
+        /// <param name="sourcePixels">The <see cref="Span{T}"/> to the source colors.</param>
+        /// <param name="destVectors">The <see cref="Span{T}"/> to the destination vectors.</param>
         internal virtual void ToScaledVector4(
             Configuration configuration,
-            ReadOnlySpan<TPixel> sourceColors,
-            Span<Vector4> destinationVectors)
+            ReadOnlySpan<TPixel> sourcePixels,
+            Span<Vector4> destVectors)
+        {
+            Guard.NotNull(configuration, nameof(configuration));
+            Guard.DestinationShouldNotBeTooShort(sourcePixels, destVectors, nameof(destVectors));
+
+            Utils.Vector4Converters.Default.DangerousToScaledVector4(sourcePixels, destVectors);
+        }
+
+        /// <summary>
+        /// Bulk version of <see cref="IPixel.FromVector4"/> converting alpha premultiplied 'sourceVectors.Length' pixels into 'destinationColors'.
+        /// </summary>
+        /// <param name="configuration">A <see cref="Configuration"/> to configure internal operations</param>
+        /// <param name="sourceVectors">The <see cref="Span{T}"/> to the source vectors.</param>
+        /// <param name="destPixels">The <see cref="Span{T}"/> to the destination colors.</param>
+        internal virtual void FromPremultipliedVector4(
+            Configuration configuration,
+            ReadOnlySpan<Vector4> sourceVectors,
+            Span<TPixel> destPixels)
+        {
+            Guard.NotNull(configuration, nameof(configuration));
+            Guard.DestinationShouldNotBeTooShort(sourceVectors, destPixels, nameof(destPixels));
+
+            Utils.Vector4Converters.Default.DangerousFromPremultipliedVector4(sourceVectors, destPixels);
+        }
+
+        /// <summary>
+        /// Bulk version of <see cref="IPixel.ToVector4()"/> converting 'sourceColors.Length' pixels into alpha premultiplied 'destinationVectors'.
+        /// </summary>
+        /// <param name="configuration">A <see cref="Configuration"/> to configure internal operations</param>
+        /// <param name="sourcePixels">The <see cref="Span{T}"/> to the source colors.</param>
+        /// <param name="destVectors">The <see cref="Span{T}"/> to the destination vectors.</param>
+        internal virtual void ToPremultipliedVector4(
```

Comment:
Unlike `ToVector4()` this is only implemented by `Rgba32.PixelOperations`. Currently, using this method in `ResizeProcessor` wires out the optimizations for the remaining "RGBA - like" pixel types (`Argb32`, `Bgra32`, `Rgb24`, `Bgr24`). (They are delegating to `Vector4Converters.RgbaCompatible.ToVector4()`).

Reply:
Ah yes, the fix should be fairly trivial then. We simply need to ensure that all the overloads we use in `Rgba32.PixelOperations` are also overloaded in our `RgbaCompatible` generated code.